arduino/src/arduino.cpp

```diff
@@ -0,0 +1,214 @@
+    if (helper != 1)
+    {
+        serial.closeDevice();
+    }
+
+    serial.closeDevice();
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+
+void Arduino::checkConnectionToConsole()
+{
+    if (checkConnection() != Arduino::ERROR_ARDUINO::SUCCSESS)
+    {
+        std::cout << "there was a problem with the connection" << std::endl;
+    }
+
+    std::cout << "the connection is successful!-" << std::endl;
+}
+
+void Arduino::addFile(const std::string &file)
+{
+    this->files.push_back(file);
+}
+
+int Arduino::checkSum(const char *message)
+{
+    int sum = 0;
+    const char *p = &message[0];
+    while (*p != NULL)
+    {
+        sum += *message;
+        ++p;
+    }
+
+    return sum;
+}
+
+Arduino::ERROR_ARDUINO Arduino::openSerial()
+{
+    int err = serial.openDevice(this->portName.c_str(), this->buadRate);
+    if (err != Arduino::ERROR_ARDUINO::SUCCSESS)
+        return Arduino::ERROR_ARDUINO::ERROR_WITH_SERIALPORT;
+    this->isOpen = true;
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+
+Arduino::ERROR_ARDUINO Arduino::serialCommend(const std::string &message)
+{
+    if (!this->isOpen)
+        return Arduino::ERROR_ARDUINO::ERROR_WITH_SERIALPORT;
+    serial.writeString(message.c_str());
+    // check sum should be added.
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+
+Arduino::ERROR_ARDUINO Arduino::writeFromBufferToFile(const std::string &data, int place)
+{
+    std::ofstream dataLog;
+    std::string fileName = FILE_NAME;
+    fileName.append(std::to_string(place));
+    fileName.append(FILE_NAME_END);
+    dataLog.open(fileName, std::ios_base::app);
+    if (!dataLog.is_open())
+    {
+        return Arduino::ERROR_ARDUINO::ERROR_WITH_FILES;
+    }
+
+    dataLog << std::time(nullptr) << std::endl;
+    dataLog << data << std::endl;
+    dataLog.close();
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+
+Arduino::ERROR_ARDUINO Arduino::getDataWithWhileLoop()
+{
+    Arduino::ERROR_ARDUINO return_code = Arduino::ERROR_ARDUINO::SUCCSESS;
+    return_code = openSerial();
+    if (return_code != Arduino::ERROR_ARDUINO::SUCCSESS)
+    {
+        goto l_cleanup;
+    }
+
+    while (true)
+    {
+        if (serial.available())
+        {
+            uint32_t num = 0;
+            this->serial.readBytes(&num, sizeof(num));
+            switch (num)
+            {
+            case Arduino::HEADERS::GET_FLOAT:
+                return_code = receiveFloatFromSensor();
+                if (return_code != SUCCSESS)
+                    goto l_cleanup;
+                continue;
+            case Arduino::HEADERS::ASK_FOR_OVERALL_STATUS:
+                return_code = checkStatusFromArduino();
+                if (return_code != SUCCSESS)
+                    goto l_cleanup;
+                continue;
+            default:
+                continue;
+            }
+        }
+    }
+
+l_cleanup:
+    return return_code;
+}
+
+Arduino::ERROR_ARDUINO Arduino::receiveFloatFromSensor()
```

Comment:
the name tells me that this function should return somehow the float that it received but it doesn't